clang-tools-extra/clang-tidy/bugprone/BranchCloneCheck.cpp

```diff
@@ -102,6 +102,210 @@ void BranchCloneCheck::registerMatchers(MatchFinder *Finder) {
       this);
   Finder->addMatcher(switchStmt().bind("switch"), this);
   Finder->addMatcher(conditionalOperator().bind("condOp"), this);
+  Finder->addMatcher(ifStmt(hasDescendant(ifStmt())).bind("ifWithDescendantIf"),
```

Comment:
`hasDescendant` will match lots of unrelated cases which will slow the speed. move the later check to here

Reply:
Hello @HerrCai0907 , I do not fully understand your suggestion. Could you describe what you mean by "move the later check to here"? Thanks for taking the time to review. Best!

Reply:
The current matcher will check for `if`s as descendants, not only as direct children. So code like
```c++
void foo() {
    if (true) {
        []() {
            if (false) {
            }
        };
    }
}
```

will still produce a match, that you filter out later, instead of in the matcher.
A matcher like `ifStmt(anyOf(has(ifStmt().bind("inner_if")), has(compoundStmt(has(ifStmt().bind("inner_if"))))).bind("ifWithDescendantIf")` will ensure that only direct children are looked at. To enforce the requirement of the first statement inside a compound statement, adding a custom matcher would be best IMO.

Reply:
Thanks @5chmidti. I think I had resolved the comment from @HerrCai0907 by narrowing the match. Would you agree that's sufficient, or would you like to see an improvement in this area before moving forward here? Thanks